File: src/main/java/com/probielab/microiot/api/HttpServerVerticle.java
package com.probielab.microiot.api;

import com.probielab.microiot.api.router.ASimpleRouter;
import com.probielab.microiot.utils.reactivex.log4vertx;
import io.vertx.reactivex.core.AbstractVerticle;
import io.vertx.reactivex.core.eventbus.EventBus;
import io.vertx.reactivex.core.http.HttpServer;
import io.vertx.reactivex.ext.web.Router;

public class HttpServerVerticle extends AbstractVerticle {
  public static final String CONFIG_HTTP_SERVER_PORT = "http.server.port";


  @Override
  public void start() {
    EventBus eb = vertx.eventBus();

    HttpServer server = vertx.createHttpServer();
    Router router = ASimpleRouter.getRouter(vertx);

    int portNumber = config().getInteger(CONFIG_HTTP_SERVER_PORT, 8080);
    server
      .requestHandler(router)
      .listen(portNumber, ar -> {
        if (ar.succeeded()) {
          log4vertx.info(eb, "Http Server running on port:" + portNumber);
        } else {
          log4vertx.error(eb, "Cannot start http server", ar.cause());
        }
      });
  }
}

File: src/main/kotlin/com/probielab/microiot/api/HttpServerVerticle.kt
package com.probielab.microiot.api

import com.probielab.microiot.api.router.ASimpleRouter
import com.probielab.microiot.utils.reactivex.log4vertx
import io.vertx.core.AsyncResult
import io.vertx.reactivex.core.AbstractVerticle
import io.vertx.reactivex.core.http.HttpServer
import java.io.ByteArrayOutputStream
import java.io.IOException
import java.io.PrintStream

class HttpServerVerticle : AbstractVerticle() {
  override fun start() {
    val eb = vertx.eventBus()
    val server = vertx.createHttpServer()
    val router = ASimpleRouter.getRouter(vertx)
    val portNumber = config().getInteger(CONFIG_HTTP_SERVER_PORT, 8080)
    server
      .requestHandler(router)
      .listen(portNumber) { ar: AsyncResult<HttpServer?> ->
        if (ar.succeeded()) {
          log4vertx.info(eb, "Http Server running on port:$portNumber")
        } else {
          val baos = ByteArrayOutputStream()
          try {
            ar.cause().printStackTrace(PrintStream(baos))
            baos.close()
          } catch (e: IOException) {
            e.printStackTrace()
          } finally {
            log4vertx.error(eb, "Cannot start http server", ar.cause())
          }
        }
      }
  }

  companion object {
    const val CONFIG_HTTP_SERVER_PORT = "http.server.port"
    const val CONFIG_MICRO_IOT_DB_QUEUE = "microiotdb.queue"
  }
}
